File: gin/repositories/user_repository.go
package repositories

import "gin-benchmark/models"

type UserRepository struct {
	users []*models.User
}

func NewUserRepository() *UserRepository {
	return &UserRepository{
		users: []*models.User{
			{Id: "1", Name: "Gianfranco", Email: "gianfranco@email.com", Password: "Test123!"},
			{Id: "2", Name: "Gianfranco2", Email: "gianfranco@email.com", Password: "Test123!"},
			{Id: "3", Name: "Gianfranco3", Email: "gianfranco@email.com", Password: "Test123!"},
			{Id: "4", Name: "Gianfranco4", Email: "gianfranco@email.com", Password: "Test123!"},
			{Id: "5", Name: "Gianfranco5", Email: "gianfranco@email.com", Password: "Test123!"},
		},
	}
}

func (r *UserRepository) GetAllUsers() []*models.User {
	return r.users
}

func (r *UserRepository) GetUserById(id string) *models.User {
	for _, user := range r.users {
		if user.Id == id {
			return user
		}
	}
	return nil
}

func (r *UserRepository) GetUserByEmail(email string) *models.User {
	for _, user := range r.users {
		if user.Email == email {
			return user
		}
	}
	return nil
}

File: phoenix/lib/ash/repositories/user_repository.ex
defmodule Ash.Repositories.UserRepository do
  alias Ash.Models.User

  @users [
    User.new("1", "Gianfranco", "gianfranco@email.com", "Test123!"),
    User.new("2", "Gianfranco2", "gianfranco@email.com", "Test123!"),
    User.new("3", "Gianfranco3", "gianfranco@email.com", "Test123!"),
    User.new("4", "Gianfranco4", "gianfranco@email.com", "Test123!"),
    User.new("5", "Gianfranco5", "gianfranco@email.com", "Test123!")
  ]

  def get_all_users do
    @users
  end

  def get_user_by_id(id) do
    Enum.find(@users, fn user -> user.id == id end)
  end

  def get_user_by_email(email) do
    Enum.find(@users, fn user -> user.email == email end)
  end
end
